### PR TITLE
Update docker

### PR DESCRIPTION
**Pull request by ryanbas21**
# Description
Docker images were out of date failing preview. This refactors the docker script to better utilize a base image. the base image just sets up the node module installs for all the other images with variables.

The images source the base image and then use a newer nginx image since the old one started failing.

Also upgrades ngx-deploy script because it needed to relax the nx peer dep. ngx-deploy previously required a hard minor of the nx library, they upgraded it recently to accept 15^ type versions.

# Type of Change

Please Delete options that are not relevant

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Definition of Done

Check all that apply

- [ ] Acceptance criteria is met.
- [ ] All tasks listed in the user story have been completed.
- [ ] Coded to standards.
- [ ] Code peer-reviewed.
- [ ] Ensure backward compatibility (special attention).
- [ ] API reference docs is updated.
- [ ] Unit tests are written.
- [ ] Integration tests are written.
- [ ] e2e tests are written.
- [ ] CI build passing on the feature branch.
- [ ] Functional spec is written/updated
- [ ] contains example code snippets.
- [ ] Change log updated.
- [ ] Documentation story is created and tracked.
- [ ] UI is completed or ticket is created.
- [ ] Demo to PO and team.
- [ ] Tech debts and remaining tasks are tracked in separated ticket(s).

## Documentation

- [ ] Acceptance criteria met
- [ ] Spell-check run
- [ ] Peer reviewed
- [ ] Proofread

### CODE REVIEW

File: packages/javascript-sdk/src/fr-auth/callbacks/index.ts [hunk 1]
```diff
@@ -22,7 +23,7 @@ class FRCallback {
   /**
    * Gets the name of this callback type.
    */
-  public getType(): string {
+  public getType(): CallbackType {
```
Comment (cerebrl) [inline packages/javascript-sdk/src/fr-auth/callbacks/index.ts:26]: Is there any side-effects from changing this? I would think this is actually not entirely correct as the function doesn't return the enum itself, but a value of the enum, yeah?

File: samples/angular-todo/src/environments/environment.ts [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
-  JOURNEY_REGISTER: '',
-  API_URL: '',
-  production: '',
+   AM_URL: 'https://openam-ryan-bas.forgeblocks.com/am',
+   REALM_PATH: 'alpha',
+   WEB_OAUTH_CLIENT: 'example-react-app',
+   JOURNEY_LOGIN: 'Login',
+   JOURNEY_REGISTER: 'Registration',
+   API_URL: 'http://localhost:9443',
+   production: 'undefined'
```
Comment (cerebrl) [inline samples/angular-todo/src/environments/environment.ts:8]: I think you have a rough space here on each line.

**Review by cerebrl — COMMENTED** [review]
Unfortunately, without more info, I can't really derive enough context to know why these changes are introduced. Can you provide some additional detail to reason for the changes?

File: .dockerignore [hunk 3]
```diff
@@ -11,4 +11,22 @@ test-results
 .devcontainer
 docker-compose.yml
 node_modules
-dist/
\ No newline at end of file
+dist/
+
+# for now we dont need this directory!
```
Comment (cerebrl) [inline .dockerignore:16]: Non-blocking, but not sure what this comment is addressing.

File: Dockerfile [hunk 4]
```diff
@@ -1,8 +1,9 @@
-FROM node:lts
+FROM okteto/node:16 as builder
```
Comment (cerebrl) [inline Dockerfile:1]: Do we want to move to Node 18, since it is now the LTS? I think 16 is now considered out-dated.

**Review by cerebrl — APPROVED** [review]

File: Dockerfile (hunk 4 above)
Comment (ryanbas21) [reply Dockerfile:1]: we can do this in another ticket, i dont wanna upgrade to 18 and retest everyhting.

**Pull request merged**